tests/h/services/search_index/_queue_test.py

```diff
@@ -137,206 +136,194 @@ def add_where(self, queue):
 
 class TestSync:
     def test_it_does_nothing_if_the_queue_is_empty(self, batch_indexer, queue):
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
     def test_it_ignores_jobs_that_arent_scheduled_yet(
-        self, add_all, batch_indexer, queue
+        self, batch_indexer, factories, now, queue
     ):
-        add_all(schedule_in=ONE_WEEK_IN_SECONDS)
+        factories.SyncAnnotationJob(scheduled_at=now + datetime_.timedelta(hours=1))
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
 
-    @pytest.mark.usefixtures("with_queued_annotations")
-    def test_it_ignores_jobs_beyond_limit(
-        self, all_annotation_ids, batch_indexer, queue
-    ):
-        queue.sync(LIMIT)
+    def test_it_ignores_jobs_beyond_limit(self, batch_indexer, factories, queue):
+        limit = 1
+        factories.SyncAnnotationJob.create_batch(size=limit + 1)
+
+        queue.sync(limit)
 
-        for annotation_id in all_annotation_ids[LIMIT:]:
-            assert annotation_id not in batch_indexer.index.call_args[0][0]
+        assert len(batch_indexer.index.call_args[0][0]) == limit
 
     def test_it_ignores_jobs_that_are_expired(
         self, batch_indexer, db_session, factories, now, queue
     ):
-        annotation = factories.Annotation()
-        queue.add_by_id(annotation.id, "test_tag", schedule_in=MINUS_5_MIN_IN_SECS)
-        # Change the job's expires_at date so that it's expired.
-        job = db_session.query(Job).one()
-        job.expires_at = now - datetime_.timedelta(minutes=1)
+        job = factories.SyncAnnotationJob(expires_at=now - datetime_.timedelta(hours=1))
 
-        counts = queue.sync(LIMIT)
+        counts = queue.sync(1)
 
         assert counts == {}
         batch_indexer.index.assert_not_called()
-        assert db_session.query(Job).all() == [job]
+        assert job in db_session.query(Job)
```

Comment:
There are noise jobs in the DB now (see below). The test just meant to assert that `job` was in the DB, it didn't really mean to assert the `job` was the _only_ job in the DB.